### PR TITLE
Add support for Jobs api v21

### PR DESCRIPTION
**Pull request by Bolshem**
@samshuster I added "v21" for conflicting classes while leaving non-conflicting with the default naming schema. Please review and tell me what you think

### CODE REVIEW

**Comment by samshuster** [comment]
Oh wow yep I didn't realize the extent of classes that would need to be created with this. Thanks for doing it.

Overall, I think I am ok with going this route, but I wonder if separating by package might be cleaner in the long run (especially after the legacy service/dtos are removed at some point far from now).

so for example com.edmunds.rest.databricks.DTO.v21.*
And then removing the v21 from the DTO class names themselves

Pros:
- For teams that only use one or the other, could be cleaner
- When eventually old Job stuff is deprecated, aren't stuck with v21 in class name at least.

Cons:
- If team needs to use both, there will be conflicts that would have to be solved by full package qualification.

What are your thoughts? And sorry I didn't think about this other option before you did all of this.

File: src/main/java/com/edmunds/rest/databricks/service/JobServiceV21.java
Comment (samshuster) [inline src/main/java/com/edmunds/rest/databricks/service/JobServiceV21.java:105]: oh ok you went ahead and kept this method in the new interface?

File: src/main/java/com/edmunds/rest/databricks/DTO/jobs/RunSubmitTaskDTO.java
Comment (samshuster) [inline src/main/java/com/edmunds/rest/databricks/DTO/jobs/RunSubmitTaskDTO.java:1]: duplicate license

File: src/main/java/com/edmunds/rest/databricks/DTO/permissions/AccessControlRequestForGroupDTO.java
Comment (samshuster) [inline src/main/java/com/edmunds/rest/databricks/DTO/permissions/AccessControlRequestForGroupDTO.java:24]: thank you for adding all of these permission and access related dto

File: src/main/java/com/edmunds/rest/databricks/service/JobServiceV21.java
Comment (Bolshem) [reply src/main/java/com/edmunds/rest/databricks/service/JobServiceV21.java:105]: Yes, as it is required for many other methods like :)

**Comment by Bolshem** [comment]
@samshuster I spent about 2 hours thinking of what would be the best way to do it.

My options were:


* Having package names like ...DTO.v21.* -> but this would end in also having ....service.v21.*, and also fully qualified names in many files that use both, and many APIs already use combinations of old and new DTOs

* Having package like ...v21.DTO.* -> but this would make a non-logical approach since many v21 APIs use classes outside of the v21 package

* Having all files in one folder, adding v21 to class names - this (in my opinion) sends a clear message of what API is used, and you could say that we add v21 in test mode) and you can do a release with a minor version update, and after some time, when all possible issues are fixed, you can do a major version (breaking change) update saying jobs v2.0 is deprecated and renaming all v21 to normal names. Since you are a maintainer, I will leave the final decision to you :) I am ok with either.

File: src/main/java/com/edmunds/rest/databricks/service/JobServiceV21.java
Comment (Bolshem) [reply src/main/java/com/edmunds/rest/databricks/service/JobServiceV21.java:105]: @samshuster This method is used by `getJobByName`, which is heavily used in tests to delete all test jobs. I had no choice.

File: src/main/java/com/edmunds/rest/databricks/DTO/jobs/RunSubmitTaskDTO.java
Comment (Bolshem) [reply src/main/java/com/edmunds/rest/databricks/DTO/jobs/RunSubmitTaskDTO.java:1]: fixed

**Comment by Bolshem** [comment]
My main thought is do you intend do keep .v21. in package name forever ? If no - then anyway at some point you will have to make a release with renaming => breaking changes. If yes, it will create an awkward situation when future releases (lets say v2.2) will use some objects from v21 and from v20, and so on resulting in some very mixed class usage.

WDYT?

File: src/main/java/com/edmunds/rest/databricks/service/JobServiceV21.java
Comment (samshuster) [reply src/main/java/com/edmunds/rest/databricks/service/JobServiceV21.java:105]: ah ok yep got it. So the LOE was higher to remove these methods after all.

**Comment by Bolshem** [comment]
btw, seems like edmunds.com sells  cars. Is it still your site?

**Comment by samshuster** [comment]
ok thanks for the write up. Definitely a tricky situation we are in. Lets go ahead and proceed with v21 in the class names like you have now. I am going to take another look over, but most likely I will be able to merge this in today. I will raise middle version for the release.

**Comment by Bolshem** [comment]
@samshuster btw, seems like edmunds.com sells cars. Is it still your site?

**Comment by samshuster** [comment]
> @samshuster btw, seems like edmunds.com sells cars. Is it still your site?

Hey yeah I still work for Edmunds which owns edmunds.com

**Review by samshuster — APPROVED** [review]
Great job and thank you so much for your contribution!

**Pull request merged**
